nodejs-env/src/test/scala/org/scalajs/jsenv/nodejs/NodeJSTest.scala

```diff
@@ -1,79 +1,14 @@
 package org.scalajs.jsenv.nodejs
 
+import org.scalajs.jsenv.JSEnv
 import org.scalajs.jsenv.test._
 
 import org.junit.Test
 import org.junit.Assert._
 
-class NodeJSTest extends TimeoutComTests {
-
-  protected def newJSEnv: NodeJSEnv = new NodeJSEnv
-
-  /** Node.js strips double percentage signs - #500 */
-  @Test
-  def percentageTest: Unit = {
-    val counts = 1 to 15
-    val argcs  = 1 to 3
-    val strings = counts.map("%" * _)
-
-    val strlists = for {
-      count  <- argcs
-      string <- strings
-    } yield List.fill(count)(string)
-
-    val codes = for {
-      strlist <- strlists
-    } yield {
-      val args = strlist.map(s => s""""$s"""").mkString(", ")
-      s"console.log($args);\n"
-    }
-
-    val result = strlists.map(_.mkString(" ") + "\n").mkString("")
-
-    codes.mkString("").hasOutput(result)
-  }
-
-  /** Node.js console.log hack didn't allow to log non-Strings - #561 */
-  @Test
-  def nonStringTest: Unit = {
-
-    """
-    console.log(1);
-    console.log(undefined);
-    console.log(null);
-    console.log({});
-    console.log([1,2]);
-    """ hasOutput
-    """|1
-       |undefined
-       |null
-       |[object Object]
-       |1,2
-       |""".stripMargin
-  }
-
-  @Test
-  def slowJSEnvTest: Unit = {
-    val com = comRunner("""
-      setTimeout(function() {
-        scalajsCom.init(function(msg) {
-          scalajsCom.send("pong: " + msg);
-        });
-      }, 1000);
-    """)
-
-    val n = 20
-
-    start(com)
-
-    for (_ <- 1 to n)
-      com.send("ping")
-
-    for (_ <- 1 to n)
-      assertEquals(com.receive(), "pong: ping")
-
-    com.close()
-    com.await(DefaultTimeout)
-  }
-
-}
+class NodeJSRunTest extends RunTests(new NodeJSEnv, withCom = false)
+class NodeJSRunComTest extends RunTests(new NodeJSEnv, withCom = true)
+class NodeJSTimeoutRunTest extends TimeoutRunTests(new NodeJSEnv, withCom = false)
+class NodeJSTimeoutRunComTest extends TimeoutRunTests(new NodeJSEnv, withCom = true)
+class NodeJSComTest extends ComTests(new NodeJSEnv)
+class NodeJSTimeoutComTest extends TimeoutComTests(new NodeJSEnv)
```

Comment:
I hate that we have to do the expansion this way, but I haven't figured out a better way. JUnit4 does not support dynamic test generation :-/

Reply:
I hacked JUnit a lot :) Now we have a way.